domain/operation/doc.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package operation defines the domain model, contracts, and documentation for
+// Juju operations.
+//
+// An operation represents a higher-level unit of work initiated by a user or
+// by Juju itself, typically comprising one or more actions/tasks executed
+// against applications, units, or machines. The operation domain provides the
+// types shared across the service and state subpackages, and serves as the
+// package-level documentation entrypoint for the operation domain.
+//
+// ## Package layout
+//   - service: high-level business API used by apiserver facades and workers to
+//     manage operations for a model. It is intentionally decoupled from the
+//     state layer.
+//   - state: access to persistence (controller/model database) relating to
+//     operations. It exposes a minimal interface consumed by the service.
+//   - types.go: DTOs used as arguments/results for state and service calls where
+//     they need to be visible to both subpackages.
+//
+// ## Conventions
+//   - Args types for state methods live in the state package.
+//   - Params types for service methods live in the top-level domain package or
+//     service package as appropriate.
+//   - Internal sqlair in/out types are private to the state package.
```

Comment:
Again, these conventions don't belong here since they are common to all domain packages. And again I'd prefer not documenting these this way.